cedar-policy-validator/src/custom_schema/ast.rs

```diff
@@ -0,0 +1,187 @@
+use cedar_policy_core::{
+    ast::{Id, Name as CName},
+    parser::{Loc, Node},
+};
+use itertools::Either;
+use miette::SourceSpan;
+use nonempty::NonEmpty;
+use smol_str::SmolStr;
+
+use super::to_json_schema::PrimitiveType;
+
+const IPADDR_EXTENSION: &str = "ipaddr";
+const DECIMAL_EXTENSION: &str = "decimal";
+pub(super) const CEDAR_NAMESPACE: &str = "__cedar";
+
+pub type Ident = Node<Id>;
```

Comment:
now that we're using `Node` for these other purposes, `Node` isn't really a great name.  Should we change the name?  It's not a public type, so that wouldn't be a breaking change.  `WithLoc`?